Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/MRI_BlockRecord/EyeLink_MRI_BlockRecord.m

```diff
@@ -0,0 +1,430 @@
+                        Eyelink('Message', 'TRIGGER_RECEIVED');
+                        blockOnset = GetSecs; % Block onset time
+                        vbl = Screen('Flip', window); % Present stimulus
+                        break;
+                    end
+                end
+            else % All subsequent trials in block
+                vbl = Screen('Flip', window, blockOnset + (trialDur*(i-1)) - 0.5*ifi); % Present stimulus. Allow half flip interval for precise flip timing)
+            end
+            % Write message to EDF file to mark the start time of stimulus presentation.
+            Eyelink('Message', 'STIM_ONSET');
+            % Write !V IMGLOAD message to EDF file: creates backdrop image for DataViewer
+            % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Image Commands
+            Eyelink('Message', '!V IMGLOAD CENTER %s %d %d', stimName, width/2, height/2);
+            % Write !V IAREA message to EDF file: creates interest area around image in DataViewer
+            % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Interest Area Commands
+            Eyelink('Message', '!V IAREA RECTANGLE %d %d %d %d %d %s', 1, round(width/2-imgInfo.Width/2), round(height/2-imgInfo.Height/2), round(width/2+imgInfo.Width/2), round(height/2+imgInfo.Height/2),'IMAGE_IA');
+            
+            % STEP 7.4: WAIT FOR STIMULUS TIMER; CLEAR SCREEN AND SHOW CROSSHAIRS
+            
+            % Prepare crosshairs for end of trial
+            Screen('DrawTexture', window, backgroundTexture); % Prepare background texture on backbuffer
+            Screen('DrawLine', window, 0, round(width/2-20), round(height/2), round(width/2+20), round(height/2), 5);
+            Screen('DrawLine', window, 0, round(width/2), round(height/2-20), round(width/2), round(height/2+20), 5);
+                % Check that eye tracker is  still recording. Otherwise close and transfer copy of EDF file to Display PC
+                err = Eyelink('CheckRecording');
+                if(err ~= 0)
+                    fprintf('EyeLink Recording stopped!\n');
+                    % Transfer a copy of the EDF file to Display PC
+                    Eyelink('SetOfflineMode');% Put tracker in idle/offline mode
+                    Eyelink('CloseFile'); % Close EDF file on Host PC
+                    Eyelink('Command', 'clear_screen 0'); % Clear trial image on Host PC at the end of the experiment
+                    WaitSecs(0.1); % Allow some time for screen drawing
+                    % Transfer a copy of the EDF file to Display PC
+                    transferFile; % See transferFile function below
+                    cleanup; % Abort experiment (see cleanup function below)
+                    return
+                end
+                Screen('Flip', window, stimDur + vbl - 0.5*ifi); % Present crosshairs. Allow half flip interval for precise flip timing)

```

Comment:
dto. see above.

Reply:
resolved in 0abf0cbe36691edea3a6d49a87b79149ad7e6f94